linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd

```diff
@@ -373,8 +373,14 @@ This section defines age classes - see code for details.
 ```{r clean_age}
 
 x <- x %>%
-  mutate(
-      age = as.numeric(gsub("_", ".", age)),
+  mutate(age = gsub("_", ".", age))
+
+# line below converts values that were initially in scientific notation (indicative of age < 1 year) to # prevent NA's from arising when calling "age = as.numeric(age)"
+  x$age[grep("E.",x$age)]=map_chr(.x=x$age[grep("E.",x$age)],.f=function(.x){return(as.character(as.numeric(substr(.x,1,regexpr("E.",.x)-1))*10^-as.numeric(substr(.x,regexpr("E.",.x)+2,nchar(.x)))))})
```

Comment:
This is hard to read. Best break this down into intermediate object, including the subsetting using grep and the function. Might be best to refer explicitely to the package for `map_chr` i.e. `purrr::map_chr` since `purrr` is not used much elsewhere. 

Style things - check the `lintr` package but in a nutshell: 

* use `<-` for assignment
* flank operators with spaces
* add space after `,`